src/hub.js

```diff
@@ -364,13 +366,33 @@ async function handleHubChannelJoined(entryManager, hubChannel, messageDispatch,
       debug: !!isDebug
     });
 
-    while (!scene.components["networked-scene"] || !scene.components["networked-scene"].data) await nextTick();
+    while (!scene.components["networked-scene"] || !scene.components["networked-scene"].data) {
+      await AFRAME.scenes[0].systems.ticker.nextTick();
+    }
 
     scene.components["networked-scene"]
       .connect()
       .then(() => {
         let newHostPollInterval = null;
 
+        let done = 0;
+        const i = window.setInterval(() => {
+          if (window.stuffToLoad === 0) {
+            done += 1;
+            if (done > 10) {
+              window.uiroot && window.uiroot.setState({ sceneObjectsLoaded: true });
+              window.clearInterval(i);
+            } else {
+              const text = `Joining lobby...`;
+              window.uiroot && window.uiroot.setState({ loadingText: text });
+            }
+          } else {
+            done = 0;
+            const text = `Loading ${window.stuffToLoad} more object${window.stuffToLoad > 1 ? "s" : ""}...`;
+            window.uiroot && window.uiroot.setState({ loadingText: text });
+          }
+        }, 100);
```

Comment:
I get the sense I would prefer this loading stuff to be in its own React component. hub.js is pretty crowded.

Reply:
(Once it becomes its own React component I will have suggestions about the state on that component and the interface for updating the loading text.)